internal/operands/common-templates/reconcile.go

```diff
@@ -36,18 +42,19 @@ func WatchClusterTypes() []operands.WatchType {
 }
 
 type commonTemplates struct {
-	templatesBundle   []templatev1.Template
-	deployedTemplates map[string]bool
+	templatesByArch map[string][]templatev1.Template
 }
 
 var _ operands.Operand = &commonTemplates{}
 
 func New(templates []templatev1.Template) operands.Operand {
-	deployedTemplates := make(map[string]bool)
-	for _, t := range templates {
-		deployedTemplates[t.Name] = true
+	templatesByArch := map[string][]templatev1.Template{}
+	for _, template := range templates {
+		templateArch := GetTemplateArch(&template)
+		templatesByArch[templateArch] = append(templatesByArch[templateArch], template)
```

Comment:
Why not dedup the slices or only add the template if it's not already in the slice? Could be as easy as calling `slices.Sort` and `slices.Compact`. Would save to implement the custom logic to create the set in `Reconcilce`.

Reply:
I think there are no duplicates in the templates.
The set that is created in `Reconcile` is used to check if a template exists, not to avoid duplicates.
I will try to make it easier to understand.

Reply:
This set is used to quickly filter templates based on architectures, so that `Reconcile` can only create compatible templates.